src/dsp_tools/commands/project/models/resourceclass.py

```diff
@@ -110,120 +110,75 @@ def ptype(self) -> Ptype:
         return self._ptype
 
     @classmethod
-    def fromJsonObj(cls, con: Connection, context: Context, jsonld_obj: Any) -> tuple[str, HasProperty]:
-        rdf = context.prefix_from_iri("http://www.w3.org/1999/02/22-rdf-syntax-ns#")
-        rdfs = context.prefix_from_iri("http://www.w3.org/2000/01/rdf-schema#")
-        owl = context.prefix_from_iri("http://www.w3.org/2002/07/owl#")
-        knora_api = context.prefix_from_iri("http://api.knora.org/ontology/knora-api/v2#")
-        salsah_gui = context.prefix_from_iri("http://api.knora.org/ontology/salsah-gui/v2#")
-
-        if jsonld_obj.get("@type") is None or jsonld_obj.get("@type") != owl + ":Restriction":
+    def fromJsonObj(cls, con: Connection, context: Context, jsonld_obj: dict[str, Any]) -> tuple[str, HasProperty]:
+        rdf_iri = context.prefix_from_iri("http://www.w3.org/1999/02/22-rdf-syntax-ns#")
+        rdfs_iri = context.prefix_from_iri("http://www.w3.org/2000/01/rdf-schema#")
+        owl_iri = context.prefix_from_iri("http://www.w3.org/2002/07/owl#")
+        knora_api_iri = context.prefix_from_iri("http://api.knora.org/ontology/knora-api/v2#")
+        salsah_gui_iri = context.prefix_from_iri("http://api.knora.org/ontology/salsah-gui/v2#")
+
+        if jsonld_obj.get("@type") is None or jsonld_obj.get("@type") != owl_iri + ":Restriction":
             raise BaseError("Expected restriction type")
 
-        #
-        # let's get the cardinality
-        #
-        cardinality: Cardinality
-        if jsonld_obj.get(owl + ":cardinality") is not None:
-            cardinality = Cardinality.C_1
-        elif jsonld_obj.get(owl + ":maxCardinality") is not None:
-            cardinality = Cardinality.C_0_1
-        elif jsonld_obj.get(owl + ":minCardinality") is not None:
-            if jsonld_obj.get(owl + ":minCardinality") == 0:
-                cardinality = Cardinality.C_0_n
-            elif jsonld_obj.get(owl + ":minCardinality") == 1:
-                cardinality = Cardinality.C_1_n
-            else:
-                raise BaseError("Problem with cardinality")
-        else:
-            raise BaseError("Problem with cardinality")
+        cardinality = cls._fromJsonObj_get_cardinality(jsonld_obj, owl_iri)
 
-        #
-        # Now let's get the property IRI
-        #
+        ontology_id, property_id, ptype = cls._fromJsonObj_get_prop_type_iri(
+            context, jsonld_obj, knora_api_iri, owl_iri, rdf_iri, rdfs_iri
+        )
+
+        gui_order: int = None
+        if jsonld_obj.get(salsah_gui_iri + ":guiOrder") is not None:
+            gui_order = jsonld_obj[salsah_gui_iri + ":guiOrder"]
+        return property_id, cls(
+            con=con,
+            context=context,
+            ontology_id=ontology_id,
+            property_id=property_id,
+            cardinality=cardinality,
+            gui_order=gui_order,
+            ptype=ptype,
+        )
+
+    @classmethod
```

Comment:
In this file we have some `@classmethod`, some `@staticmethod`, and some which are neither of the two. Have you got a system behind this?

Reply:
I followed to formalities that were suggested by the ide. I imagine other functions in the code may also be class / static methods but are not declared as such. But I did not systematically look for those, only declared the functions I touched anyway.